### PR TITLE
Move to static BK pipeline for metricbeat

### PR DESCRIPTION
**Pull request by dliappis**
## Proposed commit message

This commit refactors the metricbeat pipeline to the pipeline selector
approach as laid out in https://github.com/elastic/beats/pull/38783

## Related issues

- Relates https://github.com/elastic/ingest-dev/issues/3072#issuecomment-2046970938

## Tests

- Without any labels: https://buildkite.com/elastic/beats-metricbeat/builds/4684
- **Labels:** Using windows-2019 to trigger the corresponding extended test: https://buildkite.com/elastic/beats-metricbeat/builds/4691
- **Labels:** Using windows-10/11/2019 to trigger ALL extended windows tests: https://buildkite.com/elastic/beats-metricbeat/builds/4694
- **Labels:** Using macos: https://buildkite.com/elastic/beats-metricbeat/builds/4721#018ecc0d-3559-4852-9fcc-0caf7980f66c

- Complete run with all labels and latest commit: https://buildkite.com/elastic/beats-metricbeat/builds/4751 / https://buildkite.com/elastic/beats-metricbeat/builds/4764

    ### GH status screenshots

    #### Mandatory
    image
    image

    #### Extended Windows
    image

    #### Packaging
    image

    #### macOS (x86_64)
    image

### CODE REVIEW

File: .buildkite/metricbeat/pipeline.yml [hunk 1]
```diff
@@ -16,44 +17,261 @@ env:
...
+        artifact_paths:
+          - "metricbeat/build/*.xml"
+          - "metricbeat/build/*.json"
+        notify:
+          - github_commit_status:
+              context: "metricbeat: Windows 2022 Unit Tests"
+
+  - group: "Metricbeat Extended Windows Tests"
+    key: "metricbeat-extended-win-tests"
+    if: build.env("BUILDKITE_PULL_REQUEST") == "false" || build.env("GITHUB_PR_LABELS") =~ /.*windows.*/
```
Comment (dliappis) [inline .buildkite/metricbeat/pipeline.yml:161]: todo: make this more permissive

File: .buildkite/metricbeat/pipeline.yml [hunk 2]
```diff
@@ -16,44 +17,261 @@ env:
   IMAGE_WIN_2019: "family/platform-ingest-beats-windows-2019"
   IMAGE_WIN_2022: "family/platform-ingest-beats-windows-2022"
 
-  #Packaging
+  # Integration Tests
+  K8S_VERSION: "v1.29.0"
+  ASDF_KIND_VERSION: "0.20.0"
+
+  # Packaging
   PACKAGING_ARM_PLATFORMS: "linux/arm64"
```
Comment (dliappis) [inline .buildkite/metricbeat/pipeline.yml:25]: move to packaging

File: .buildkite/metricbeat/pipeline.yml [hunk 3]
```diff
@@ -16,44 +17,261 @@ env:
...
-    command: ".buildkite/scripts/generate_metricbeat_pipeline.sh"
-    agents:
-      image: "docker.elastic.co/ci-agent-images/platform-ingest/buildkite-agent-beats-ci:latest"
-    notify:
-      - github_commit_status:
-          context: "${BEATS_PROJECT_NAME}: Load dynamic pipeline's steps"
+    # with PRs, we want to run packaging only if mandatory tests succeed
+    # for other cases, e.g. merge commits, we want to run packaging (and publish) independently of other tests
+    # this allows building DRA artifacts even if there is flakiness in mandatory tests
+    if: build.env("BUILDKITE_PULL_REQUEST") != "false"
```
Comment (dliappis) [inline .buildkite/metricbeat/pipeline.yml:244]: also honor Packaging label see https://github.com/elastic/beats/pull/38825/files#diff-18f9ca90b8c92c28fd6c97266f3b8214cbf81cdb37ed36784930965cdc8bf13eR195

File: .buildkite/pipeline.yml [hunk 4]
```diff
@@ -70,6 +70,30 @@ steps:
...
+          diff: "git diff --name-only ${GITHUB_PR_TARGET_BRANCH}...HEAD"
+          watch:
+            - path:
+                - metricbeat/
+                - .buildkite/metricbeat/
+                - .buildkite/scripts
+                #OSS
+                - go.mod
+                - pytest.ini
+                - dev-tools/**
```
Comment (alexsapran) [inline .buildkite/pipeline.yml:85]: I see in other PRs that this was adjusted to 

```suggestion
                - dev-tools/
```

@pazone can give some insight into why; better to keep it consistent.

**Review by alexsapran — APPROVED** [review]

**Pull request merged**